site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,455 @@
+```
+
+--------------------------
+
+## Authoring a Carvel Package
+
+To create a package, we need to create two Custom Resources (CRs). We will go through step by step to author an nginx helm chart:
+
+**1. Create PackageMetadata CR**: Package Metadata contains high level information and description about the package. Multiple versions of a package share same package metadata.
+
+Save the below PackageMetadata CR to a file named `pkgMetadata.yaml`
+```yaml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  # This will be the name of our package metadata
+  name: nginx.bitnami.vmware.com
+spec:
+  displayName: "Bitnami Nginx Carvel Package"
+  longDescription: "Proxifying Server"
+  shortDescription: "Proxifying Server"
+  categories:
+  - proxy-server
+  providerName: VMWare
+  maintainers:
+  - name: "Carvel"
+  - name: "CarvelInd"
+  - name: "Rohit Aggarwal"
+```
+
+**2. Package Helm Chart**: The Package CR is config including metadata and references to OCI artifacts that informs the package manager what software it holds and how to install itself onto a Kubernetes cluster.
+
+Save the below Package CR to a file named `1.0.0.yaml`
+```yaml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  # Must be of the form '<spec.refName>.<spec.version>' (Note the period)
+  name: nginx.bitnami.vmware.com.1.0.0
+spec:
+  # The name of the PackageMetadata associated with this version
+  # Must be a valid PackageMetadata name (see PackageMetadata CR for details)
+  # Cannot be empty
+  refName: nginx.bitnami.vmware.com
+  # Package version; Referenced by PackageInstall;
+  # Must be valid semver (required)
+  # Cannot be empty
+  version: 1.0.0
+  # Version release notes (optional; string)
+  releaseNotes:
+    Initial release of the nginx package by wrapping helm Chart. Nginx Helm chart version is 9.5.4
+  # valuesSchema can be used to show template values that
+  # can be configured by users when a Package is installed.
+  # These values should be specified in an OpenAPI schema format. (optional)
+  # For helm chart, we can either define the configurable values here or let it be. Even if we dont define them here, we can still customize them.
+  valuesSchema:
+    openAPIv3:
+      title: nginx.bitnami.vmware.com
+      examples:
+      properties:
+  template:
+    spec:
+      fetch:
+      - helmChart:
+          name: nginx
+          version: 9.5.4
+          repository:
+            # From where to pull the helm chart
+            url: https://charts.bitnami.com/bitnami 
+      template:
+      - helmTemplate: {}
+      deploy:
+      - kapp: {}
+```
+**Note**: This is one way of packaging helm chart. Another way is to use `imgpkg` bundle to store the helm chart itself and then reference it. 
+
+**3. Create Package Repository**: A package repository is a collection of packages and their metadata. We will use `imgpkg` bundle to create package repository.
```

Comment:
grammar nit:
"`imgpkg`" => "an imgpkg"
"create package repository." => "create a package repository."

Reply:
I think it is fine

Reply:
I would insist on these too, grammarly surfaces article usage warnings as well

Reply:
Adopted